### DISCUSSION TITLE
Store photos for notes in the camera memory

### DISCUSSION THREAD

**Discussion by OpenGreenStreet** [streetcomplete/StreetComplete#2396] [Ideas]
I can't find the photos for notes in the camera memory. It would be very good if these were stored in the DCIM area.

(de: Fotos für Notizen im Speicher der Kamera ablegen: Die Fotos für Notizen kann ich im Speicher der Kamera nicht finden. Es wäre sehr gut, wenn diese im DCIM-Bereich gespeichert werden würden.)

**Comment by matkoniecz** [comment]
Is there some use of that? Why it would be beneficial?

See https://github.com/streetcomplete/StreetComplete/issues/1162 where some things were mentioned, but maybe there is also some other use

**Reply by OpenGreenStreet** [reply]
I would like to publish some of the photos on Wikimedia Commons at a later date.

**Comment by Cj-Malone** [comment]
Similarly adding the location data to the uploaded images would save me some time.

Sometimes an image I've used in a note has some significance so when I close the note I download the image and re upload it to Mapillary, since it has no gps data I have to manually position it.

**Reply by smichel17** [reply]
I think this might be done intentionally for privacy reasons? Although it does not make a difference for location since it is attached to an OSM note.

**Reply by westnordost** [reply]
yes, all metadata is stripped for privacy reasons

**Reply by Cj-Malone** [reply]
The location is already explicitly linked due to the note, but the data protection issues are presumably the real issue. Holding that data on your server comes with the liability.

**Reply by smichel17** [reply]
Fair point. I guess if the option to also save notes to the phone's storage were added, then the location metadata could be saved there.

**Reply by mnalis** [reply]
- related discussion: https://github.com/streetcomplete/StreetComplete/issues/2880

**Comment by hb0nd** [comment]
how about a switch in settings to enable this? (off by default)

**Reply by OpenGreenStreet** [reply]
This would be a very good compromise

**Comment by deevroman** [comment]
Yesterday I came across the fact that when creating a note, something happened to SC and I lost the attached photos and had to do them again. I have SC when opening the camera does not allow using photos from the gallery, so saving photos directly would not help. But I was glad not to take a photo again, but by adding information with my hands through another editor, looking at photos from the gallery.

Also, my camera can take photos in the size of 4600×2400, and as I understand it, SC compresses photos to 1280×1280. But that was enough for now.

A few more times my notes were closed without squeezing all the information out of the photo :) Fortunately, I had time to download the photo before auto-deletion.

---

On the other hand, I understand that all these are infrequent cases, and the new setting may be of little use to anyone.

As a possible compromise, maybe by a long click on the button to send a note, save a photo to your phone? Or save them always, but only by a long click, DO NOT delete them when sending.

---

It is also possible that this feature is more to StreetComplete expert edition :) https://github.com/Helium314/StreetComplete

**Reply by westnordost** [reply]
I think it would be fine to also store all photos made with StreetComplete locally in some private or public folder. However, I am not sure if StreetComplete would then need an additional permission (local file access) if it stored the photos in a publicly readable folder. This would be unfortunate and IMO not worth the additional permission.

**Reply by OpenGreenStreet** [reply]
I would very much like to see local storage of photos. (Most of the time I try hard to take photos that are as expressive as possible. Then when I want to use them further for Wikimedia Commons, I start all over again).

Could it possibly be the other way around: within the note I can access photos from the Camera (local storage). (I start a note and add an already taken photo. Gladly turned off by default).

**Reply by hb0nd** [reply]
> However, I am not sure if StreetComplete would then need an additional permission (local file access) if it stored the photos in a publicly readable folder. This would be unfortunate and IMO not worth the additional permission.

idk if this would work, but maybe using an app-space folder & sharing them with the share dialog wouldn't need any permissions?
even an appdata folder would be good (for those who have root access)

**Reply by smichel17** [reply]
> Could it possibly be the other way around: within the note I can access photos from the Camera (local storage)

Technically, possible. However, not desired. See the links from https://github.com/streetcomplete/StreetComplete/issues/3014#issuecomment-872949131

**Reply by Helium314** [reply]
> idk if this would work, but maybe using an app-space folder & sharing them with the share dialog wouldn't need any permissions?
even an appdata folder would be good (for those who have root access)

Storing on external storage (which is the name Android devs chose for "normal" phone storage, in `Android/data` folder) would allow other apps to read it, though only up to Android 11 or 12.
Saving files in another accessible place can be done without permission only for single files (I think), i.e. you would need to store each photo separately.
